cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+  val expressionEval = new WorkflowEngineFunctions(executionDir)
+
+  /**
+    * Prepare the task and context for execution.
+    */
+  override def prepare(): Unit = {
+    logger.debug(s"Creating execution folder: $executionDir")
+    executionDir.toString.toFile.createIfNotExists(true)
+
+    try {
+      val command = initiateCommand()
+      logger.debug(s"Creating bash script for executing command: ${command}.")
+      writeBashScript(command, executionDir)
+      subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+        subs => subs.subscriber ! new TaskStatus(Status.Created, None))
+    } catch {
+      case ex: Exception => subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+        subs => subs.subscriber ! new TaskStatus(Status.Failed, Some(FailureResult(ex, None, ""))))
+    }
+  }
+
+  /**
+    * Stops a task execution.
+    */
+  override def stop(): Unit = {
+    processAbortFunc.get.apply()
+    subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+      subs => subs.subscriber ! new TaskStatus(Status.Canceled, None))
+  }
+
+  /**
+    * Executes task in given context.
+    */
+  override def execute(): Unit = {
+    subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+      subs => subs.subscriber ! executeTask)
+  }
+
+  /**
+    * Performs a cleanUp after the task was executed.
+    */
+  override def cleanUp(): Unit = {
+    // Do nothing.
+  }
+
+  /**
+    * Subscribe to events on backend.
+    */
+  override def subscribeToEvent[T](subscription: Subscription[T]): Unit = {
+    val sub = subscription.asInstanceOf[Subscription[ActorRef]]
+    subscriptions += sub
+    sub.subscriber ! Subscribed
+  }
+
+  /**
+    * Unsubscribe to events on backend.
+    */
+  override def unsubscribeToEvent[T](subscription: Subscription[T]): Unit = {
+    val sub = subscription.asInstanceOf[Subscription[ActorRef]]
+    subscriptions -= sub
+    sub.subscriber ! Unsubscribed
+  }
+
+  /**
+    * Get file content hash.
+    * @param files List of files to compute.
+    * @return List of hashes related to files.
+    */
+  override def computeInputFileHash(files: List[Path]): Map[Path, String] = {
+    files.map(path => path -> File(path.toString).md5).toMap[Path, String]
+  }
+
+  /**
+    * Get container image hash.
+    * @param imageName Image name.
+    * @return A hash.
+    */
+  override def computeContainerImageHash(imageName: String): Map[String, String] = ???
+
+  /**
+    * Gather Docker image name from runtime attributes. It it's not present returns none.
+    * @param runtimeAttributes Runtime requirements for the task.
+    */
+  private def getDockerImage(runtimeAttributes: Map[String, String]): Option[String] = {
+    if (runtimeAttributes.filter(p => p._1.equals(DockerFlag)).size == 1 &&
+      !runtimeAttributes.filter(p => p._1.equals(DockerFlag)).values.head.isEmpty) {
```

Comment:
It's not null because the the check on the previous line. 

However, since this is returning Option[String] anyway, a map would probably be even nicer.